src/main/java/com/example/demo/repository/book/BookSpecificationBuilder.java

```diff
@@ -0,0 +1,37 @@
+package com.example.demo.repository.book;
+
+import com.example.demo.dto.BookSearchParametersDto;
+import com.example.demo.model.Book;
+import com.example.demo.repository.SpecificationBuilder;
+import com.example.demo.repository.SpecificationProviderManager;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@RequiredArgsConstructor
+@Component
+public class BookSpecificationBuilder implements SpecificationBuilder<Book> {
+    private static final String AUTHOR_KEY = "author";
+    private static final String ISBN_KEY = "isbn";
+    private static final String TITLE_KEY = "title";
```

Comment:
better make these constants public and reuse in providers. You will have one place for changes in the future